brave/brave-core
# Migrate legacy cosmetic filters

Resolves https://github.com/brave/brave-browser/issues/9581

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

### Migration

Using a _previous_ browser version:
1. Visit brave://adblock in a new tab
2. Add some text of your choice to the "Custom filters" box
3. Visit a website of your choice
4. Right-click on a page element of your choice
5. Select "Brave > Block element via selector"
6. Confirm the block
7. Repeat steps 3-6 a few times to block a variety of items through the legacy system. Block multiple items on the same page at least once. Keep track of what is being blocked.

Then, testing the same profile _after_ the upgrade:
1. Visit brave://adblock in a new tab
2. Verify that any text previously added to the "Custom filters" box is still present
3. Verify that new lines have been added to the "Custom filters" box, including one corresponding to each item previously blocked using the right click menu
4. Revisit some of the websites and confirm that any previously blocked items are still hidden

### Adding new filters

Just on the new version:
1. Visit a website of your choice
2. Right-click on a page element of your choice
3. Select "Brave > Block element via selector"
4. Confirm the block
5. Verify that the blocked item is hidden, and remains hidden after a page refresh
6. Reopen or refresh brave://adblock and verify that a new line has been added corresponding to the newly blocked item
7. Repeat steps 1-6 at least one more time
